src/main/java/org/generationcp/middleware/dao/dms/StockDao.java

```diff
@@ -292,5 +305,119 @@ public void replaceExperimentStocks(final Integer oldStockId, final Integer newS
+					entryName, entryName + "_variableId",
+					entryName, entryName + "_value"
+				));
+			}
+
+			sqlQuery.append(" FROM stock s "
+				+ "       LEFT JOIN ims_lot l ON l.eid = s.dbxref_id and l.status = " + LotStatus.ACTIVE.getIntValue()
+				+ "       LEFT JOIN ims_transaction it ON l.lotid = it.lotid "
+				+ "       LEFT JOIN cvterm c ON c.cvterm_id = l.scaleid "
+				+ "       LEFT JOIN stockprop sp ON sp.stock_id = s.stock_id "
+				+ "       LEFT JOIN cvterm cvterm_variable ON cvterm_variable.cvterm_id = sp.type_id "
+				+ "WHERE s.project_id = :studyId ");
+
+
+			if (studyEntrySearchDto.getFilter()!=null) {
+				if (studyEntrySearchDto.getFilter().getEntryNumbers()!=null && !studyEntrySearchDto.getFilter().getEntryNumbers().isEmpty()) {
+					sqlQuery.append(" AND s.uniquename in (:entryNumbers)" );
+				}
+				if (studyEntrySearchDto.getFilter().getEntryIds()!=null && !studyEntrySearchDto.getFilter().getEntryIds().isEmpty()) {
+					sqlQuery.append(" AND s.stock_id in (:entryIds)" );
+				}
+			}
+			sqlQuery.append(" GROUP BY s.stock_id ");
+			GenericDAO.addPageRequestOrderBy(sqlQuery, pageable);
+
+			final SQLQuery query = this.getSession().createSQLQuery(sqlQuery.toString());
+			query.addScalar("entryId", new IntegerType());
+			query.addScalar("entryNumber", new IntegerType());
+			query.addScalar("gid", new IntegerType());
+			query.addScalar("designation", new StringType());
+			query.addScalar("entryCode", new StringType());
+			query.addScalar("lots", new IntegerType());
+			query.addScalar("available", new StringType());
+			query.addScalar("unit", new StringType());
+			for (final MeasurementVariable entryDescriptor : studyEntrySearchDto.getEntryDescriptors()) {
+				final String entryName = entryDescriptor.getName();
+				query.addScalar(entryName + "_propertyId", new IntegerType());
+				query.addScalar(entryName + "_variableId", new IntegerType());
+				query.addScalar(entryName + "_value", new StringType());
+			}
+
+			query.setParameter("studyId", studyEntrySearchDto.getStudyId());
+			if (studyEntrySearchDto.getFilter()!=null) {
+				if (studyEntrySearchDto.getFilter().getEntryNumbers()!=null && !studyEntrySearchDto.getFilter().getEntryNumbers().isEmpty()) {
+					query.setParameterList("entryNumbers", studyEntrySearchDto.getFilter().getEntryNumbers());
+				}
+				if (studyEntrySearchDto.getFilter().getEntryIds()!=null && !studyEntrySearchDto.getFilter().getEntryIds().isEmpty()) {
+					query.setParameterList("entryIds", studyEntrySearchDto.getFilter().getEntryIds());
+				}
+			}
+			query.setResultTransformer(AliasToEntityMapResultTransformer.INSTANCE);
+			GenericDAO.addPaginationToSQLQuery(query, pageable);
+
+			final List<Map<String, Object>> results = query.list();
+			final List<StudyEntryDto> studyEntryDtos = new ArrayList<>();
+			for (final Map<String, Object> row : results) {
+				final StudyEntryDto studyEntryDto = new StudyEntryDto();
```

Comment:
Provide a constructor in `StudyEntryDto` with the expected set of fields (entryid, gid, ...,etc, ... unit)